### DISCUSSION TITLE
Panache and Test on Command Mode Applications

### DISCUSSION THREAD

**Discussion by lecogiteur** [quarkusio/quarkus#39338] [Community]
Hi,

I have created a simple command mode applications. I just insert elements in database (postgres) using PanacheEntity.
It works well. No problem.

I try to create a test but it seems I can't use the PanacheEntity in the test method.

The Entity

```java
@Entity
public class Person extends PanacheEntity {
    public String name;
}
```

The Main

```java
@QuarkusMain
public class PersonMain implements QuarkusApplication {
  @Override
  @Transactional
  public int run(String... args) {
      Person l = new Person();
      l.name = "aaa";
      l.persist();
      return 0;
  }
}
```

The test

```java
@QuarkusMainTest
class PersonMainTest {

  @Test
  @Launch
  @TestTransaction
  public void aTest() {
    Person.count();
  }
}
```
When I execute the test, it insert the person in database (I have checked with my sql client in database). But after, the test fails with this error (when it executes the test method content):
```shell
java.lang.IllegalStateException: This method is normally automatically overridden in subclasses: did you forget to annotate your entity with @Entity?

	at io.quarkus.hibernate.orm.panache.common.runtime.AbstractJpaOperations.implementationInjectionMissing(AbstractJpaOperations.java:443)
	at io.quarkus.hibernate.orm.panache.PanacheEntityBase.count(PanacheEntityBase.java:591)
```

But 

If I use `@QuarkusTest` I can execute `Person.count()` in test but the application is not launched so it's useless

**EDIT:**

This code works well. But I need to launch application manually
```java
@QuarkusTest
class PersonMainTest {

  @Test
  @TestTransaction
  public void aTest() {
    new ParsonMain().run();
    Person.count();
  }
}
```
Why I can't use @QuarkusMainTest with PanacheEntity ? Or how to use it with JPA?

Someone can help me?

**Comment by lecogiteur** [comment]
Need Help please :)

**Reply by lecogiteur** [reply]
I'm reading this thread: https://groups.google.com/g/quarkus-dev/c/HGZhNw6jsLg

It seems there is no CDI context shared like using @QuarkusTest  for @QuarkustIntegrationTest (so in @QuarkusTest we can access to context and execute request using panache). I think it's the same thing for @QuarkusMainTest . I think panache and entity manager aren't configured too in this case.

So my question, I need to test what is inserted in my database when we use @QuarkusMainTest, @QuarkustIntegrationTest, @QuarkustMainIntegrationTest. How to do that simply? Is there a simple way to do thaht or I must to use JDBC or assertj-DB for example?

**Comment by ChristianKernDev** [comment]
Is there any workaround available for this issue?

What is the reason for CDI not being available in @QuarkusMainTest?
